### DISCUSSION TITLE
Unloading issue

### DISCUSSION THREAD

**Discussion by PatCode000** [jeffreylanters/react-unity-webgl#399] [Help Wanted!]
Hello!

Everything works fine, but for about two weeks I can’t solve one problem. 
The problem occurs when I am navigating away from the page, and it seems after all the game still works behind th scene. My guess is that it wasn’t properly unloaded. It’s only terminated when page is refreshed or navigated in old fashioned way.

I tried to use react’s useEffect hook to recognize when user is leaving the page (component is removed) but when I do, unload() function tells that unity game wasn’t instantiated (the game music still plays in the background anyway).

I wouldn’t like to solve this problem the tricky way by using seperate page for the game as purpose of React is to work as a single page application.
Thank you for all the answers in advance

![289610766_5333789066676811_8154019476793339839_n-2](https://user-images.githubusercontent.com/51359385/174965656-39c78fea-3862-4bb3-87ed-52e71a5733e4.jpg)
.

**Comment by hananbenshabat** [comment]
I also have a similar issue. Does it give you the error when using development build? it solved it for me.

**Comment by jeffreylanters** [comment]
Hi, unloading problems are a known issue in later Unity versions #250. This is a problem with the actual webgl build, so there is not much we can do except with for unity to fix it.

**Reply by ShawnCentra** [reply]
I just ran into the same issue.

**Reply by ShawnCentra** [reply]
User clicks open -> modal opens -> renders unity object 

When the modal closes and the user tries to open the modal again -> I get the same error as above. 

I was thinking removing the canvas might be the solution when the modal closes but I haven't had luck on removing it properly.

**Reply by ShawnCentra** [reply]
Would you recommend reverting to a previous build for now until unity fixes the issue? Ie. react-unity-webgl 8.8.0  and Unity 2019 instead of 9.0 and 2021?

**Reply by ShawnCentra** [reply]
Fixed with the version reversal.  https://github.com/jeffreylanters/react-unity-webgl/discussions/412

**Reply by jeffreylanters** [reply]
> Would you recommend reverting to a previous build for now until unity fixes the issue? Ie. react-unity-webgl 8.8.0 and Unity 2019 instead of 9.0 and 2021?

The issue is not related to the version of react-unity-webgl but rather to the version of Unity of which the build is made with. So you can use any version of the module, just keep in mind that if you want to truly unload your Unity application, you'll have to use an older version of Unity in order to create that build until Unity releases a fix.

**Comment by industry2205** [comment]
i solved null check at _JS_DOM_MapViewportCoordinateToElementLocalCoordinate in .framework.js.

in my case, variable canvas was null when unity page in or out.
so i just added 'if' to canvas.
